src/utilities/sequence-editor/to-seq-json.ts

```diff
@@ -70,9 +84,147 @@ export async function sequenceToSeqJson(
+    steps.push({
+      args: [],
+      stem: 'UNKNOWN',
+      time: { tag: 'UNKNOWN', type: 'ABSOLUTE' },
+      type: 'command',
+    });
+  }
+
+  // ground epoch
+  return {
+    description,
+    ground_epoch,
+    metadata,
+    name,
+    steps: steps as [Step, ...Step[]],
+    time,
+    type: 'request',
+  };
+}
+
+function parseGroundBlockEvent(stepNode: SyntaxNode, text: string): GroundBlock | GroundEvent {
+  const time = parseTime(stepNode, text);
+
+  const nameNode = stepNode.getChild('GroundName');
+  const name = nameNode ? unquoteUnescape(text.slice(nameNode.from, nameNode.to)) : 'UNKNOWN';
+
+  const argsNode = stepNode.getChild('Args');
+  // step not in dictionary, so not passing command dict
+  const args = argsNode ? parseArgs(argsNode, text, null, name) : [];
+
+  const description = parseDescription(stepNode, text);
+  const metadata = parseMetadata(stepNode, text);
+  const models = parseModel(stepNode, text);
+
+  return {
+    args,
+    description,
+    metadata,
+    models,
+    name,
+    time,
+    type: stepNode.name === 'GroundBlock' ? 'ground_block' : 'ground_event',
+  };
+}
+
+function parseActivateLoad(stepNode: SyntaxNode, text: string): Activate | Load {
+  const time = parseTime(stepNode, text);
+
+  const nameNode = stepNode.getChild('SequenceName');
+  const sequence = nameNode ? unquoteUnescape(text.slice(nameNode.from, nameNode.to)) : 'UNKNOWN';
+
+  const argsNode = stepNode.getChild('Args');
+  // step not in dictionary, so not passing command dict
+  const args = argsNode ? parseArgs(argsNode, text, null, sequence) : [];
+
+  const engine = parseEngine(stepNode, text);
+  const epoch = parseEpoch(stepNode, text);
+
+  const description = parseDescription(stepNode, text);
+  const metadata = parseMetadata(stepNode, text);
+  const models = parseModel(stepNode, text);
+
+  return {
+    args,
+    description,
+    engine,
+    epoch,
+    metadata,
+    models,
+    sequence,
+    time,
+    type: stepNode.name === 'Load' ? 'load' : 'activate',
+  };
+}
+
+function parseEngine(stepNode: SyntaxNode, text: string): number | undefined {
+  const engineNode = stepNode.getChild('Engine')?.getChild('Number');
+  return engineNode ? parseInt(text.slice(engineNode.from, engineNode.to), 10) : undefined;
+}
+
+function parseEpoch(stepNode: SyntaxNode, text: string): string | undefined {
+  const epochNode = stepNode.getChild('Epoch')?.getChild('String');
+  return epochNode ? unquoteUnescape(text.slice(epochNode.from, epochNode.to)) : undefined;
+}
+
+function parseStep(child: SyntaxNode, text: string, commandDictionary: CommandDictionary | null): Step | null {
+  switch (child.name) {
+    case 'Command':
+      return parseCommand(child, text, commandDictionary);
+    case 'Activate':
+    case 'Load':
+      return parseActivateLoad(child, text);
+    case 'GroundBlock':
+    case 'GroundEvent':
+      return parseGroundBlockEvent(child, text);
+  }
+  // SeqN allows comments that don't map to Seq.JSON
```

Comment:
is this specific to only SeqN or potentially other grammars?

Reply:
I expect SASF/SATF to have an analog

Reply:
Can we make this comment more generic to indicate that this potentially allows for more things to be converted to seq json other than just SeqN?

Reply:
pushed